sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashTest.java

```diff
@@ -0,0 +1,128 @@
+ * Licensed under the MIT License.
+ *
+ */
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.InternalObjectNode;
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.routing.PartitionKeyInternal;
+import com.azure.cosmos.models.*;
+import com.azure.cosmos.rx.TestSuiteBase;
+import com.azure.cosmos.util.CosmosPagedIterable;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CosmosMultiHashTest extends TestSuiteBase {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private String preExistingDatabaseId = CosmosDatabaseForTest.generateId();
+    private CosmosClient client;
+    private CosmosDatabase createdDatabase;
+    private CosmosContainer createdMultiHashContainer;
+
+    @Factory(dataProvider = "clientBuilders")
+    public CosmosMultiHashTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void before_CosmosItemTest() {
+
+        client = getClientBuilder().buildClient();
+        createdDatabase = createSyncDatabase(client, preExistingDatabaseId);
+        String collectionName = UUID.randomUUID().toString();
+
+        PartitionKeyDefinition partitionKeyDefinition = new PartitionKeyDefinition();
+        partitionKeyDefinition.setKind(PartitionKind.MULTI_HASH);
+        partitionKeyDefinition.setVersion(PartitionKeyDefinitionVersion.V2);
+        ArrayList<String> paths = new ArrayList<>();
+        paths.add("/city");
+        paths.add("/zipcode");
+        partitionKeyDefinition.setPaths(paths);
+
+        CosmosContainerProperties containerProperties = getCollectionDefinition(collectionName, partitionKeyDefinition);
+
+        //MultiHash collection create
+        CosmosContainerResponse containerResponse = createdDatabase.createContainer(containerProperties);
+
+        //MultiHash collection read
+        createdMultiHashContainer = createdDatabase.getContainer(collectionName);
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        logger.info("starting ....");
+        //MultiHash Collection delete
+        createdMultiHashContainer.delete();
+        safeDeleteSyncDatabase(createdDatabase);
+        safeCloseSyncClient(client);
+    }
+
+    @Test(groups = { "simple" }, timeOut = TIMEOUT)
+    public void itemCRUDTest() throws Exception {
```

Comment:
code style:
we don't use "Test" suffix nor prefix. please remove the Test suffix.

```suggestion
    public void itemCRUD() throws Exception {
```